src/target/km1m7xx.c

```diff
@@ -0,0 +1,1965 @@
+		return ERROR_FAIL;
+	}
+
+	cortex_m->common_magic = CORTEX_M_COMMON_MAGIC;
+	cortex_m->apsel = pc->ap_num;
+
+	cortex_m_init_arch_info(target, cortex_m, pc->dap);
+
+	return ERROR_OK;
+}
+
+COMMAND_HANDLER(km1m7xx_handle_calc_image_checksum_command)
+{
+	uint8_t			*buffer;
+	size_t			buf_cnt;
+	uint32_t		image_size;
+	struct image	image;
+	int				i;
+	int 			retval;
+	uint32_t		checksum = 0;
+
+	if (CMD_ARGC != 1)
+		return ERROR_COMMAND_SYNTAX_ERROR;
+
+	image.base_address_set	= 0;
+	image.base_address		= 0;
+	image.start_address_set	= 0;
+
+	retval = image_open(&image, CMD_ARGV[0], NULL);
+	if (retval != ERROR_OK)
+		return retval;
+
+	image_size = 0x0;
+	retval = ERROR_OK;
+	for (i = 0; i < image.num_sections; i++) {
+		buffer = malloc(image.sections[i].size);
+		if (buffer == NULL) {
+			command_print(CMD,
+					"error allocating buffer for section (%d bytes)",
+					(int)(image.sections[i].size));
+			break;
+		}
+		retval = image_read_section(&image, i, 0x0, image.sections[i].size, buffer, &buf_cnt);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		/* calculate checksum of image */
+		retval = image_calculate_checksum(buffer, buf_cnt, &checksum);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		free(buffer);
+		image_size += buf_cnt;
+	}
+	return retval;
+}
+
+COMMAND_HANDLER(km1m7xx_handle_calc_memory_checksum_command)
+{
+	uint8_t			*buffer;
+	size_t			buf_cnt;
+	uint32_t		image_size;
+	struct image	image;
+	int				i;
+	int 			retval;
+	uint32_t		checksum = 0;
+	uint32_t		mem_checksum = 0;
+	struct target	*target = get_current_target(CMD_CTX);
+
+	if (CMD_ARGC != 1)
+		return ERROR_COMMAND_SYNTAX_ERROR;
+
+	if (!target) {
+		LOG_ERROR("no target selected");
+		return ERROR_FAIL;
+	}
+
+	image.base_address_set	= 0;
+	image.base_address		= 0;
+	image.start_address_set	= 0;
+
+	retval = image_open(&image, CMD_ARGV[0], NULL);
+	if (retval != ERROR_OK)
+		return retval;
+
+	image_size = 0x0;
+	retval = ERROR_OK;
+	for (i = 0; i < image.num_sections; i++) {
+		buffer = malloc(image.sections[i].size);
+		if (buffer == NULL) {
+			command_print(CMD,
+					"error allocating buffer for section (%d bytes)",
+					(int)(image.sections[i].size));
+			break;
+		}
+		retval = image_read_section(&image, i, 0x0, image.sections[i].size, buffer, &buf_cnt);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		/* calculate checksum of image */
+		retval = image_calculate_checksum(buffer, buf_cnt, &checksum);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		retval = target_checksum_memory(target, image.sections[i].base_address, buf_cnt, &mem_checksum);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		free(buffer);
+		image_size += buf_cnt;
+	}
+
+	return retval;
+}
+
+COMMAND_HANDLER(km1m7xx_handle_keycode_file_command)
+{
+	FILE		*fp_keyfile;
+	char		key_str[16];
+	int			key_count;
+
+	if (CMD_ARGC != 1) {
+		return(ERROR_COMMAND_SYNTAX_ERROR);
```

Comment:
Remove parentheses unless an expression is returned.

Reply:
Please end with a new line.

Reply:
Please refer to the following answer:
https://unix.stackexchange.com/a/18789

Reply:
The surrounding lines end with a new line ('/n' = 0x0a).
Where is the problem?

Reply:
Please ignore the second comment. Thanks.

Reply:
I've got it. I will fix it for the first comment.

Reply:
Removed parentheses.